### ISSUE TITLE
sinatra: `caller_locations': wrong number of arguments (given 2, expected 0) (ArgumentError)

### ISSUE THREAD

**Issue by deepj** [oracle/truffleruby#1934]
```
truffleruby 20.1.0-dev-d6e1b291, like ruby 2.6.5, GraalVM CE Native [x86_64-darwin]
```

This is minimal reproducible code extracted from [ruby_event_store-browser](https://github.com/RailsEventStore/rails_event_store/tree/master/ruby_event_store-browser). The problem lies in https://github.com/sinatra/sinatra/blob/master/lib/sinatra/base.rb#L1514 It's in conflict with `Kernel.caller_locations`. No issue under MRI though.

To reproduce:

```
gem install sinatra
rackup
```

**config.ru**

```ruby
require 'sinatra/base'

class MyApp < Sinatra::Base
  def self.for(host: nil, path: nil, environment: :production)
    self.tap do |app|
      app.settings.instance_exec do
        set :public_folder, "#{__dir__}/../../../public"
      end
    end
  end
end

run MyApp.for
```
Error:

```
~/.gem/truffleruby/2.6.5/gems/sinatra-2.0.8.1/lib/sinatra/base.rb:1514:in `caller_locations': wrong number of arguments (given 2, expected 0) (ArgumentError)
	from ~/dev/truffleruby/bugs/sinatra/config.ru:8:in `__dir__'
	from ~/dev/truffleruby/bugs/sinatra/config.ru:8:in `block (2 levels) in for'
	from ~/dev/truffleruby/bugs/sinatra/config.ru:7:in `instance_exec'
	from ~/dev/truffleruby/bugs/sinatra/config.ru:7:in `block in for'
	from ~/dev/truffleruby/bugs/sinatra/config.ru:6:in `tap'
	from ~/dev/truffleruby/bugs/sinatra/config.ru:6:in `for'
```

**Comment by norswap** [comment]
Minimal reprod:
```ruby
p __dir__ # ok
def caller_locations
  'override'
end
p __dir__ # nope
```

**Comment by norswap** [comment]
Fixed in 1de99bf4631adcd326857855beadb921a5d2475e
